testcases/crypto/ec_func.c

```diff
@@ -1387,7 +1387,8 @@ CK_RV run_GenerateECCKeyPairSignVerify()
         if (rc != CKR_OK) {
             if (rc == CKR_MECHANISM_PARAM_INVALID ||
                 rc == CKR_ATTRIBUTE_VALUE_INVALID ||
-                rc == CKR_CURVE_NOT_SUPPORTED) {
+                rc == CKR_CURVE_NOT_SUPPORTED ||
+                rc == CKR_FUNCTION_CANCELED) {
```

Comment:
please see my mail about how to fix the CKR_FUNCTION_CANCELED case.

Reply:
Done. Routine rsa_ec_generate_keypair now checks if the curve is supported, before calling the ep11lib, and returns CKR_CURVE_NOT_SUPPORTED if not. This even makes local variable curve_type and local subroutine get_curve_type_from_attrs superfluous.